src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -1135,6 +1137,36 @@ void fightAirRaidsAndStrategicBombing(final IDelegateBridge delegateBridge, Supp
     }
   }
 
+  /**
+   * 'Auto-fight' defenseless battles.
+   */
+  public void fightAutoKills(final IDelegateBridge bridge) {
+    // Kill undefended transports. Done first to remove potentially dependent sea battles
+    // Which could block amphibious assaults below
+    final GameData gameData = bridge.getData();
+    for (final Territory territory : getPendingBattleSites(false)) {
+      final IBattle battle = getPendingBattle(territory, false, BattleType.NORMAL);
+      final List<Unit> defenders = new ArrayList<>();
+      defenders.addAll(battle.getDefendingUnits());
+      final List<Unit> sortedUnitsList = new ArrayList<>(Match.getMatches(defenders,
+                 Matches.UnitCanBeInBattle(true, !territory.isWater(), gameData, 1, false, true, true)));
+      Collections.sort(sortedUnitsList, new UnitBattleComparator(false, ProData.unitValueMap,
+          TerritoryEffectHelper.getEffects(territory), gameData, false, false));
+      Collections.reverse(sortedUnitsList);
+      if (DiceRoll.getTotalPower(
+          DiceRoll.getUnitPowerAndRollsForNormalBattles(sortedUnitsList, defenders, false, false, gameData,
+            territory, TerritoryEffectHelper.getEffects(territory), false, null), gameData) == 0) {
+        battle.fight(bridge);
+      }
+    }
+    getPendingBattleSites(false).stream().map(territory -> getPendingBattle(territory, false, BattleType.NORMAL))
+         .forEach( battle -> {
+           if (battle instanceof NonFightingBattle && getDependentOn(battle).isEmpty()) {
```

Comment:
Can I ask why? Wouldn't that be a dummy method, returning true for being a NonFightingBattle or similar? We'd have to avoid DependentBattle to avoid serialisation failures wouldn't we?

And doesn't such a move cause different methods of determining battle type are scattered across the code.